doc/advopt.txt

```diff
@@ -45,14 +45,17 @@ Advanced options:
   --spellSuggest|:num       show at most `num >= 0` spelling suggestions on typos.
                             if `num` is not specified (or `auto`), return
                             an implementation defined set of suggestions.
-  -w:on|off|list, --warnings:on|off|list
-                            turn all warnings on|off or list all available
-  --warning[X]:on|off       turn specific warning X on|off
-  --hints:on|off|list       turn all hints on|off or list all available
-  --hint[X]:on|off          turn specific hint X on|off
-  --warningAsError[X]:on|off
+  --hints:on|off|all|none|list
+                            `on|off` enables or disables currently selected hints and acts
+                            as a reversible gate, whereas `all|none` selects or deselects
+                            all hints. `list` reports which hints are selected.
```

Comment:
What are the "currently selected hints"? What's a "reversible gate"? Maybe you mean:

```
--hints:default|nodefault|none|all|list

```

(And then on/off are mapped to default/nodefault for backward compatibility.)

Reply:
I've pushed a commit to improve the description a bit, but further suggestions welcome.

> What are the "currently selected hints"?

hints selected by previously processed flags/configuration; different flags affect hints eg `--verbosity`, `hint:X` in user config, some other cmdline flags which also enable/disable hints, etc. Eg, user may have --hint:X:off in user config but re-enable it in cmdline.

> What's a "reversible gate"?

this is pre-existing behavior. hints:on|off affects `optHints`:
```nim
proc hasHint*(conf: ConfigRef, note: TNoteKind): bool =
  # THIS IS A BIT SIMPLIFIED
  optHints in conf.options and note in conf.notes

# ditto with warnings:on|off:
proc hasWarn*(conf: ConfigRef, note: TNoteKind): bool =
  optWarns in conf.options and note in conf.notes
```

it's reversible because if user has `hints:off` somewhere in their config (or cmdline), a `hints:on` parsed after will reverse the effect of `hints:off`, flipping `optHints` back on, and restoring the selected hints.

This is also used in code with pragmas:
```
ansi_c.nim:14:8:{.push hints:off, stack_trace: off, profiler: off.}
...
{.push warnings:off.}
```

> Maybe you mean:

that sounds like it'd have different semantics that the already pre-existing semantics: in particular it's unclear what `nodefault` would mean. 

Again, this PR doesn't change semantics of on|off; it just adds 2 new options which are useful (especially in tests where you want to test 1 thing at a time).